### PR TITLE
XCLOUD Cryptomining

### PR DESCRIPTION
**Pull request by melamedbn**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3728)

## Description
A new playbook for handling XCLOUD Cryptomining alerts.

## Screenshots
Cloud_Cryptomining

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml [hunk 1]
```diff
@@ -0,0 +1,2049 @@
...
+    isautoswitchedtoquietmode: false
+  "27":
+    id: "27"
+    taskid: b7c92358-4c34-42ab-844a-87cc9cbac517
+    type: regular
+    task:
+      id: b7c92358-4c34-42ab-844a-87cc9cbac517
+      version: -1
+      name: Delete the instance
+      description: Deletes the specified instance resource. For more information, see Stopping or Deleting an Instance.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml:815]: Where is Stopping and Deleting an Instance?  
From where can I get more information?

**Comment by ShirleyDenkberg** [comment]
@melamedbn 
- On line 1313 in playbook-Account_Enrichment_-_Generic_v2.2.yml, is that description correct? Should it be "The account ID"?
- On line 815 in playbook-Cloud_Response_-_GCP.yml it says:
        Deletes the specified instance resource. For more information, see **Stopping or Deleting an Instance.**
Where is **Stopping or Deleting an Instance**? It's not in the yml file.
- Also on line 882
For more information, see **Stopping an instance**.

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Response_-_Generic.yml [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+    "paper": {
+      "dimensions": {
+        "height": 725,
+        "width": 1300,
+        "x": -240,
+        "y": 60
+      }
+    }
+  }
+inputs: []
```
Comment (altmannyarden) [inline Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Response_-_Generic.yml:482]: shouldnt the inputs from the subplaybooks be here?

File: Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.2.yml [hunk 3]
```diff
@@ -1066,7 +1257,7 @@ outputs:
 - contextPath: IdentityIQ.Identity.emails
   description: Array of email objects.
 - contextPath: IdentityIQ.Identity.entitlements
-  description: Array of entitlements objects that the identity has.
+  description: Array of entitlement objects that the identity has.
```
Comment (altmannyarden) [inline Packs/CommonPlaybooks/Playbooks/playbook-Account_Enrichment_-_Generic_v2.2.yml:1260]: @melamedbn please add the MS output to the playbook outputs as well

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_Azure.yml [hunk 4]
```diff
@@ -0,0 +1,1397 @@
...
+      - "30"
+    scriptarguments:
+      resource_group:
+        complex:
+          root: alertJson._all_events
+          filters:
+          - - operator: containsGeneral
+              left:
+                value:
+                  simple: alertJson._all_events.referenced_resource_name
```
Comment (altmannyarden) [inline Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_Azure.yml:581]: Should update the playbook with the generic flow

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_AWS.yml [hunk 5]
```diff
@@ -0,0 +1,1911 @@
+id: Cloud Response - AWS
+version: -1
+name: Cloud Response - AWS
+description: |-
+  This playbook provides response actions to Azure. The following are available for execution automatically/manually:
```
Comment (altmannyarden) [inline Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_AWS.yml:5]: should be AWS and not Azure :)

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_Azure.yml [hunk 6]
```diff
@@ -0,0 +1,1397 @@
...
+      IP:
+        complex:
+          root: blockIndicatorsList
+          transformers:
+          - operator: SetIfEmpty
+            args:
+              applyIfEmpty: {}
+              defaultValue:
+                value:
+                  simple: alertJson._all_events.caller_ip
```
Comment (altmannyarden) [inline Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_Azure.yml:748]: here as well

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_AWS.yml [hunk 7]
```diff
@@ -0,0 +1,1911 @@
...
+          - operator: SetIfEmpty
+            args:
+              applyIfEmpty: {}
+              defaultValue:
+                value:
+                  simple: inputs.resourceName
+                iscontext: true
+      region:
+        complex:
+          root: alertJson._all_events
```
Comment (altmannyarden) [inline Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_AWS.yml:598]: shouldnt we use the generic way?

File: Packs/Azure-Enrichment-Remediation/pack_metadata.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            "mandatory": true,
+            "display_name": "Azure Compute"
+        },
+        "CommonScripts": {
+            "mandatory": true,
+            "display_name": "Common Scripts"
+        },
+        "MicrosoftGraphUser": {
+            "mandatory": true,
+            "display_name": "Microsoft Graph User"
```
Comment (altmannyarden) [inline Packs/Azure-Enrichment-Remediation/pack_metadata.json:30]: Should it be mandatory?

File: Packs/CommonTypes/IncidentFields/incidentfield-cloudresourcelist.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    "neverSetAsRequired": false,
+    "isReadOnly": false,
+    "useAsKpi": false,
+    "locked": false,
+    "system": false,
+    "content": true,
+    "group": 0,
+    "hidden": false,
+    "openEnded": true,
+    "associatedTypes": [
```
Comment (altmannyarden) [inline Packs/CommonTypes/IncidentFields/incidentfield-cloudresourcelist.json:21]: need to add a new type you created as well

File: Packs/CommonTypes/IncidentFields/incidentfield-cloudregionlist.json [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    "isReadOnly": false,
+    "useAsKpi": false,
+    "locked": false,
+    "system": false,
+    "content": true,
+    "group": 0,
+    "hidden": false,
+    "openEnded": true,
+    "associatedTypes": [
+        "Cortex XDR - XCLOUD"
```
Comment (altmannyarden) [inline Packs/CommonTypes/IncidentFields/incidentfield-cloudregionlist.json:22]: need to add the new type you created as well

File: Packs/CortexXDR/IncidentTypes/incidenttype-Cortex_XDR_-_XCLOUD_Cryptomining.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
+{
+    "id": "Cortex XDR - XCLOUD Cryptomining",
+    "version": -1,
+    "vcShouldIgnore": false,
+    "locked": false,
+    "name": "Cortex XDR - XCLOUD Cryptomining",
+    "prevName": "Cortex XDR - XCLOUD Cryptomining",
+    "color": "#98DCFF",
+    "playbookId": "Cortex XDR incident handling v3",
```
Comment (altmannyarden) [inline Packs/CortexXDR/IncidentTypes/incidenttype-Cortex_XDR_-_XCLOUD_Cryptomining.json:9]: why this playbook?

File: Packs/GCP-Enrichment-Remediation/README.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
+There are multiple GCP content packs for multiple GCP products (GSuiteAdmin, GCP-IAM, Google Cloud Compute, etc).  The intent was that users can install and use only the packs they need.  However, if a GCP playbook uses multiple pack integrations (such 
+as Compute and MSGraphUsers), they can't reside in one of the current packs because they include content from multiple packs.  This 
+pack was created as a place to put GCP playbooks that use GCP integrations from multiple packs with a focus on enrichment and 
+remediation.
```
Comment (altmannyarden) [inline Packs/GCP-Enrichment-Remediation/README.md:4]: Cant understand the pack description

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP_README.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
+This playbook provides response actions to Azure, the following are available for execution automatically/manually:
```
Comment (altmannyarden) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP_README.md:1]: GCP and not Azure :)

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml [hunk 14]
```diff
@@ -0,0 +1,2049 @@
...
+      {
+        "position": {
+          "x": -750,
+          "y": 1660
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
```
Comment (altmannyarden) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml:297]: If we are not checking if the integration is enabled before the test, let's change it to true (let's verify it across all the response playbooks

File: Packs/AWS-Enrichment-Remediation/README.md [hunk 15]
```diff
@@ -1,24 +1,26 @@
-There are multiple AWS content packs for multiple AWS products (EC2, IAM, Route53, etc).  The intent of this was so that users can install and use only the packs they need.  However, in that case that an AWS playbook uses multiple pack integartions (such as EC2 and IAM), they can't reside in one of the current packs because they include content from multiple.  This pack was created as a place to but AWS playbooks that use AWS integrations from multiple packs with a focus on enrichment and remediation.
```
Comment (altmannyarden) [inline Packs/AWS-Enrichment-Remediation/README.md:1]: You deleted this section on purpose?

File: Packs/CortexXDR/Playbooks/Cortex_XDR_Alerts_Handling.yml [hunk 16]
```diff
@@ -378,15 +393,183 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "12":
+    id: "12"
+    taskid: 18ea3ba1-78ee-4ce3-844b-44853df600da
+    type: condition
+    task:
+      id: 18ea3ba1-78ee-4ce3-844b-44853df600da
+      version: -1
+      name: Is XCLOUD Cryptomining alert available?
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/Cortex_XDR_Alerts_Handling.yml:519]: what do you mean by available?

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml (hunk 1 above)
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml:815]: As mentioned in Slack, those are OOTB command descriptions

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml [hunk 17]
```diff
@@ -0,0 +1,2049 @@
...
+    isautoswitchedtoquietmode: false
+  "28":
+    id: "28"
+    taskid: 434c2cb3-871e-4e21-8035-bd26a4287da4
+    type: regular
+    task:
+      id: 434c2cb3-871e-4e21-8035-bd26a4287da4
+      version: -1
+      name: Stop the instance
+      description: Stops a running instance, shutting it down cleanly, and allows you to restart the instance at a later time. Stopped instances do not incur VM usage charges while they are stopped. However, resources that the VM is using, such as persistent disks and static IP addresses, will continue to be charged until they are deleted. For more information, see Stopping an instance.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml:882]: As mentioned in Slack, those are OOTB command descriptions

File: Packs/Azure-Enrichment-Remediation/pack_metadata.json (hunk 8 above)
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/pack_metadata.json:30]: It should in this playbook, the all intent is to respond

File: Packs/GCP-Enrichment-Remediation/README.md (hunk 12 above)
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/README.md]: reviewed.

File: Packs/CortexXDR/Playbooks/Cortex_XDR_incident_handling_v3.yml [hunk 18]
```diff
@@ -1770,7 +1725,100 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "79":
+    id: "79"
+    taskid: 3ba84f9e-aaf6-4bd3-8d22-cfc00499e075
+    type: condition
+    task:
+      id: 3ba84f9e-aaf6-4bd3-8d22-cfc00499e075
+      version: -1
+      name: Check for Cryptomining alert type
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/Cortex_XDR_incident_handling_v3.yml:1792]: For future alerts, lets change to something like should continue with incident investigation? or something more generic

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP_README.md (hunk 13 above)
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP_README.md:1]: fixed.

File: Packs/AWS-Enrichment-Remediation/README.md (hunk 15 above)
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/README.md:1]: just moved it, still there.

File: Packs/CortexXDR/Playbooks/Cortex_XDR_Alerts_Handling.yml (hunk 16 above)
Comment (melamedbn) [reply Packs/CortexXDR/Playbooks/Cortex_XDR_Alerts_Handling.yml:519]: will change it to "Was XCLOUD Cryptomining alert found?"

Comment (melamedbn) [reply Packs/CortexXDR/Playbooks/Cortex_XDR_Alerts_Handling.yml]: done

File: Packs/CortexXDR/Playbooks/playbook-Cloud_Enrichment.yml [hunk 19]
```diff
@@ -0,0 +1,962 @@
...
+    simple: "True"
+  required: false
+  description: Determines whether to convert the IP address to a hostname using a DNS query (True/ False).
+  playbookInputQuery:
+- key: InternalRange
+  value: {}
+  required: false
+  description: A list of internal IP ranges to check IP addresses against. \nFor IP Enrichment - Generic v2 playbook.
+  playbookInputQuery:
+outputs: []
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/playbook-Cloud_Enrichment.yml:959]: maybe we should also add playbook outputs

File: Packs/CortexXDR/Playbooks/playbook-Cloud_Cryptomining.yml [hunk 20]
```diff
@@ -0,0 +1,935 @@
...
+
+          Cortex XSOAR has detected activity related to cryptomining.
+          After careful analysis, Cortex XSOAR found the activity malicious.
+
+          Please enter Cortex XSOAR incident ID ${incident.id} for further investigation and response.
+      subject:
+        simple: Cryptominig activity detected by Cortex XSOAR
+      to:
+        complex:
+          root: inputs.SOCEmailAddress
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/playbook-Cloud_Cryptomining.yml:201]: what happens if the input is empty? lets check that first

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml (hunk 14 above)
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml]: fixed

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml [hunk 21]
```diff
@@ -0,0 +1,2049 @@
...
+          - operator: SetIfEmpty
+            args:
+              applyIfEmpty: {}
+              defaultValue:
+                value:
+                  simple: inputs.resourceName
+                iscontext: true
+      zone:
+        complex:
+          root: alertJson._all_events
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Response_-_GCP.yml:837]: fixed

File: Packs/CommonTypes/IncidentFields/incidentfield-cloudresourcelist.json (hunk 9 above)
Comment (melamedbn) [reply Packs/CommonTypes/IncidentFields/incidentfield-cloudresourcelist.json:21]: fixed.

File: Packs/CommonTypes/IncidentFields/incidentfield-cloudregionlist.json (hunk 10 above)
Comment (melamedbn) [reply Packs/CommonTypes/IncidentFields/incidentfield-cloudregionlist.json]: fixed

File: Packs/CortexXDR/IncidentTypes/incidenttype-Cortex_XDR_-_XCLOUD_Cryptomining.json (hunk 11 above)
Comment (melamedbn) [reply Packs/CortexXDR/IncidentTypes/incidenttype-Cortex_XDR_-_XCLOUD_Cryptomining.json:9]: fixed.

File: Packs/CortexXDR/Playbooks/playbook-Cloud_Cryptomining.yml (hunk 20 above)
Comment (melamedbn) [reply Packs/CortexXDR/Playbooks/playbook-Cloud_Cryptomining.yml]: added comment in the playbook's input

File: Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_Cloud_Cryptomining.yml [hunk 22]
```diff
@@ -212,7 +212,7 @@ tasks:
     note: false
     timertriggers: []
     ignoreworker: false
-    skipunavailable: false
+    skipunavailable: true
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_Cloud_Cryptomining.yml:215]: shouldn't this be false?

Comment (melamedbn) [reply Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_Cloud_Cryptomining.yml:215]: No. we've mentioned in the input description that it will be skipped.

**Review by altmannyarden — APPROVED** [review]

File: Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_Cloud_Cryptomining.yml (hunk 22 above)
Comment (altmannyarden) [reply Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_Cloud_Cryptomining.yml:215]: yes yes you are right :)

**Comment by ShirleyDenkberg** [comment]
@altmannyarden Doc review completed.

**Pull request merged**